includes/Feed/FeedProductFormatter.php

```diff
@@ -0,0 +1,217 @@
+<?php
+namespace SkyVerge\WooCommerce\Facebook\Feed;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+use SkyVerge\WooCommerce\Facebook\Products;
+
+/**
+ * Class responsible for formatting product for feed file.
+ */
+class FeedProductFormatter {
+
+	const MAX_IMAGES_PER_PRODUCT = 5;
+
+	/**
+	 * Assembles product payload in feed upload for initial sync.
+	 *
+	 * @see FeedDataExporter::attribute_variants for explanation why $attribute_variants are passed as reference.
+	 * @since 2.6.0
+	 * @param \WC_Facebook_Product $woo_product WooCommerce product object normalized by Facebook.
+	 * @param array                $attribute_variants Array of variants attributes.
+	 * @return string product feed line data
+	 */
+	public function prepare_product_for_feed( $woo_product, &$attribute_variants ) {
+
+		$product_data  = $woo_product->prepare_product( null, \WC_Facebook_Product::PRODUCT_PREP_TYPE_FEED );
+		$item_group_id = $product_data['retailer_id'];
+
+		// Prepare variant column for variable products.
+		$product_data['variant'] = '';
+
+		if ( $woo_product->is_type( 'variation' ) ) {
```

Comment:
I'd consider moving this whole conditional block to a separate method due to it's size. It would also provide a little more clarity as to what all this code is actually doing. E.g. `::prepare_product_variation_for_feed()`.

Reply:
Refactored.